src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif

```diff
@@ -74,42 +82,153 @@
           "formattedRuleMessage": {
             "formatId": "default",
             "arguments": [
-              "n"
+              "p"
             ]
           },
+          "suppressionStates": [ "suppressedExternally" ],
+          "baselineState": "existing",
+          "level": "error",
+          "snippet": "add_core(p, offset, val);\n    return;",
           "locations": [
             {
               "analysisTarget": {
-                "uri": "collections/list.cpp",
+                "uri": "file:///home/build/src/collections/list.cpp"
+              },
+              "resultFile": {
+                "uri": "file:///home/build/src/collections/list.h",
                 "region": {
-                  "startLine": 15
+                  "startLine": 15,
+                  "startColumn": 9,
+                  "endLine": 15,
+                  "endColumn": 10,
+                  "length": 1,
+                  "offset": 254
+                }
+              },
+              "fullyQualifiedLogicalName": "collections::list:add",
+              "decoratedName": "?add@list@collections@@QAEXH@Z"
+            }
+          ],
+          "relatedLocations": [
+            {
+              "message": "\"count\" was declared here.",
```

Comment:
We've discussed this. The reason we didn't was that we didn't know where to put the message formats (it wasn't obvious that they belonged in rule.messageFormats, since they weren't necessarily rule-specific. Until we have a use case, I'd wait on this.

---
In reply to: [72280081](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72280081) [](ancestors = 72280081)